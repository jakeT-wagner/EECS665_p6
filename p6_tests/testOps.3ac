[BEGIN GLOBALS]
b
e
f
g
c
[END GLOBALS]
[BEGIN fn LOCALS]
a (formal arg of 8)
d (local var of 8 bytes)
q (local var of 8 bytes)
varTmp0 (tmp var of 8 bytes)
varTmp1 (tmp var of 8 bytes)
varTmp2 (tmp var of 8 bytes)
varTmp3 (tmp var of 8 bytes)
varTmp4 (tmp var of 8 bytes)
varTmp5 (tmp var of 8 bytes)
varTmp6 (tmp var of 8 bytes)
varTmp7 (tmp var of 8 bytes)
varTmp8 (tmp var of 8 bytes)
varTmp9 (tmp var of 8 bytes)
varTmp10 (tmp var of 8 bytes)
varTmp11 (tmp var of 8 bytes)
varTmp12 (tmp var of 8 bytes)
varTmp13 (tmp var of 8 bytes)
[END fn LOCALS]
fun_fn:     enter fn
            getarg 1 [a]
            [varTmp0] := 10 ADD64 1
            [d] := [varTmp0]
            [varTmp1] := 12 SUB64 3
            [a] := [varTmp1]
            [varTmp2] := 8 MULT64 4
            [b] := [varTmp2]
            [varTmp3] := 12 DIV64 3
            [d] := [varTmp3]
            [varTmp4] := 6 EQ64 6
            [e] := [varTmp4]
            [varTmp5] := 7 NEQ64 8
            [f] := [varTmp5]
            [varTmp6] := 1 OR64 1
            [e] := [varTmp6]
            [varTmp7] := 0 AND64 0
            [f] := [varTmp7]
            [varTmp8] := 5 LT64 6
            [e] := [varTmp8]
            [varTmp9] := 6 GT64 7
            [f] := [varTmp9]
            [varTmp10] := 5 LTE64 6
            [e] := [varTmp10]
            [varTmp11] := 6 LTE64 7
            [f] := [varTmp11]
            [varTmp12] := NEG64 5
            [d] := [varTmp12]
            [varTmp13] := NOT8 1
            [e] := [varTmp13]
lbl_0:      leave fn

