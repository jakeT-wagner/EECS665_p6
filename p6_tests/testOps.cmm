int b;
bool e;
bool f;
int g;
int fn(int a){
    int d;
    bool q;
    d = 10 + 1;
    a = 12-3;
    b = 8*4;
    d = 12/3;
    e = 6 == 6;
    f = 7 != 8;
    e = true or true;
    f = false and false;
    e = 5 < 6;
    f = 6 > 7;
    e = 5 <= 6;
    f = 6 <= 7;
    d = -5;
    e = !(true);
}
int c;