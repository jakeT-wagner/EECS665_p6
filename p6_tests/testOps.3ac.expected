[BEGIN GLOBALS]
b
e
f
g
c
[END GLOBALS]
[BEGIN fn LOCALS]
a (formal arg of 8)
d (local var of 8 bytes)
q (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
tmp2 (tmp var of 8 bytes)
tmp3 (tmp var of 8 bytes)
tmp4 (tmp var of 8 bytes)
tmp5 (tmp var of 8 bytes)
tmp6 (tmp var of 8 bytes)
tmp7 (tmp var of 8 bytes)
tmp8 (tmp var of 8 bytes)
tmp9 (tmp var of 8 bytes)
tmp10 (tmp var of 8 bytes)
tmp11 (tmp var of 8 bytes)
tmp12 (tmp var of 8 bytes)
tmp13 (tmp var of 8 bytes)
[END fn LOCALS]
fun_fn:     enter fn
            getarg 1 [a]
            [tmp0] := 10 ADD64 1
            [d] := [tmp0]
            [tmp1] := 12 SUB64 3
            [a] := [tmp1]
            [tmp2] := 8 MULT64 4
            [b] := [tmp2]
            [tmp3] := 12 DIV64 3
            [d] := [tmp3]
            [tmp4] := 6 EQ64 6
            [e] := [tmp4]
            [tmp5] := 7 NEQ64 8
            [f] := [tmp5]
            [tmp6] := 1 OR64 1
            [e] := [tmp6]
            [tmp7] := 0 AND64 0
            [f] := [tmp7]
            [tmp8] := 5 LT64 6
            [e] := [tmp8]
            [tmp9] := 6 GT64 7
            [f] := [tmp9]
            [tmp10] := 5 LTE64 6
            [e] := [tmp10]
            [tmp11] := 6 LTE64 7
            [f] := [tmp11]
            [tmp12] := NEG64 5
            [d] := [tmp12]
            [tmp13] := NOT64 1
            [e] := [tmp13]
lbl_0:      leave fn
